### PR TITLE
Judge a scan against a song the choir has already sung

### PR DESCRIPTION
**Pull request by eerovil**
Adds reviewed OMR references built from songs the choir has already sung and recorded. Cleaned scores are imploded back into their printed staff shape, then used for fixture creation, scan evaluation, and side-by-side reports.

The reviewed manifest now chooses the exact PDF and cleaned score for every consumer and verifies both recorded hashes before either file is used. A changed or missing reviewed source stops the workflow and requires manifest regeneration and another review. Reference voice totals also exclude staves hidden because they rest through the whole system.

Implosion now preserves each `lyricsSystemMap` measure range. Voices can share a printed staff in one system and split onto separate staves in a later system; unused fixed positions contain measure rests and are hidden by MuseScore. When a fixed printed position changes source staff, the incoming staff's inherited clef, key, and time signature are written at the boundary.

## Checked

- Focused implosion, reference-manifest, and real clean-flow tests: 31 passed
- All current manifest entries match both recorded hashes
- CI runs the full suite

AgentDeck session link unavailable in this worker environment.

Closes #138

### CODE REVIEW

**Comment by eerovil** [comment]
/claude Two actionable correctness issues in head `bd03fec902eff161f1fcc4f6061ff9868f3e9bd4`:

1. The reviewed manifest is not actually enforced when building/evaluating references. `fixtures/omr-songs.json` records the exact `pdf`, `cleaned`, `pdf_sha256` and `cleaned_sha256`, but `make_stem_fixture.py`, `scan_vs_reference.py`, `song_pages.py` and `implode_report.py` rediscover `*_cleaned.mscx` with `sorted(glob(...))[0]`, and none of the consumers verifies either hash. If a song folder later contains another cleaned score or either reviewed source is edited/re-cleaned, the workflow can silently judge homr against different music while still presenting the old `review.status=ok`. Please make the reviewed manifest the source of truth for the chosen files and fail loudly when their hashes no longer match (or regenerate/re-review explicitly).

2. `scan_vs_reference.reference_systems()` counts `voices` for every imploded staff before checking whether that staff has any noteheads in the system. A staff hidden because it rests through the whole system therefore contributes at least one XML `<voice>` even though `staves` correctly excludes it as unprinted. That makes a correct scan look like it has too few voices on exactly the variable-staff systems this machinery is meant to measure. Only add a staff's voice count when that staff is actually printed (`here > 0`), and add a regression test with a fully resting/hidden staff for one system.

**Comment by eerovil** [comment]
**Restarting this card's worker** (1 of 2): its turn reached its own end (success) but it never reported, so there is no report of what it reached — but a fresh worker can go and look at what is there, which is better than the card waiting for a person to. `claude_code:main` is over its usage ceiling, so this attempt asks for another account.

Its queued instructions and any worktree it registered are kept, and the new worker is asked to check what is already done before repeating it, then report. If this many attempts all end without a report the card goes to **Needs More Info** for a human.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
I found a clean inherited branch with no issue-specific commits and no PR from `agent/issue-138`. I’ll inspect the existing reference-work PR and recent history, make the manifest choose and verify the reviewed files in one shared path, fix hidden-staff voice counting, add focused regressions, and validate before updating or opening the appropriate PR.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
/claude One additional correctness blocker remains in head `f46a382943b492d526805e58a6ced9c30d70c110`: `implode._from_system_map()` collapses `lyricsSystemMap` into one global grouping, but `lyricsSystemMap` exists specifically because the printed grouping can change per system. The current algorithm therefore cannot reproduce a page where voices share a staff in one system and split onto separate staves in another.

There is already a concrete repo example in `test_clean_flow`: one range maps printed staff 1 to output `[1,2]`, while a later range maps printed staff 1 to `[1]` and printed staff 2 to `[2]`. `_from_system_map()` unions 1 and 2 because they shared once, returns one `PrintedStaff([1,2])`, and `implode()` keeps them merged for the whole score. In the later system the reference then has one tenor staff where the page has two, so fixture/scan evaluation can mark a correct scan wrong.

Please preserve the per-system grouping in the imploded reference (or otherwise encode system-specific staff splits) instead of reducing `lyricsSystemMap` to a whole-score union. Add a regression covering the existing `[1,2]` -> later `[1]` + `[2]` transition and assert the rendered/reference structure follows both systems correctly.

**Comment by eerovil** [comment]
**Restarting this card's worker** (2 of 2): it was killed mid-turn (completed), so there is no report of what it reached — but a fresh worker can go and look at what is there, which is better than the card waiting for a person to.

Its queued instructions and any worktree it registered are kept, and the new worker is asked to check what is already done before repeating it, then report. If this many attempts all end without a report the card goes to **Needs More Info** for a human.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
The inherited fix is present at `f46a382`, the worktree is clean, PR #138 points at that exact commit, and both CI jobs are green. I’ll register the PR for this refreshed generation and submit the prepared final review packet; no further repository change is needed.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
The new blocker is valid: unioning `lyricsSystemMap` cannot reproduce a later split of voices that previously shared a staff. I’m tracing the existing map contract, then I’ll preserve its measure ranges in the imploded MSCX by routing voices to fixed printed-position staves per system, add the concrete `[1,2]` then `[1]` + `[2]` regression, run focused tests, and update the existing PR.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
Review surface: source +1668−0; tests +360−0; docs +0−0; generated +319−0. Read first: `src/clean_score/implode.py`, `scripts/reference_manifest.py`, `scripts/scan_vs_reference.py`.

Beyond the card: nothing

<details><summary><code>scripts/reference_manifest.py</code> +42−0</summary>

```diff
diff --git a/scripts/reference_manifest.py b/scripts/reference_manifest.py
new file mode 100644
index 0000000..d6c3935
--- /dev/null
+++ b/scripts/reference_manifest.py
@@ -0,0 +1,42 @@
+"""Resolve and verify the files approved as OMR references."""
+
+import hashlib
+import json
+from dataclasses import dataclass
+from pathlib import Path
+
+MANIFEST = Path("fixtures/omr-songs.json")
+SONGS = Path("songs")
+
+
+@dataclass(frozen=True)
+class ReferenceFiles:
+    pdf: Path
+    cleaned: Path
+
+
+def manifest() -> dict:
+    return json.loads(MANIFEST.read_text())["songs"]
+
+
+def _verified(slug: str, entry: dict, kind: str) -> Path:
+    path = SONGS / slug / entry[kind]
+    if not path.is_file():
+        raise FileNotFoundError(f"{slug}: reviewed {kind} is missing: {path}")
+    actual = hashlib.sha256(path.read_bytes()).hexdigest()[:16]
+    expected = entry[f"{kind}_sha256"]
+    if actual != expected:
+        raise ValueError(
+            f"{slug}: reviewed {kind} changed: {path} has {actual}, expected "
+            f"{expected}; regenerate fixtures/omr-songs.json and review it again"
+        )
+    return path
+
+
+def reference_files(slug: str) -> ReferenceFiles:
+    """Return the exact reviewed files, refusing stale manifest entries."""
+    entry = manifest()[slug]
+    return ReferenceFiles(
+        pdf=_verified(slug, entry, "pdf"),
+        cleaned=_verified(slug, entry, "cleaned"),
+    )
```

</details>

<details><summary><code>scripts/make_stem_fixture.py</code> +122−0</summary>

```diff
diff --git a/scripts/make_stem_fixture.py b/scripts/make_stem_fixture.py
new file mode 100644
index 0000000..1e67595
--- /dev/null
+++ b/scripts/make_stem_fixture.py
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+"""Cut one printed system out of a song and write it as a homr stem fixture.
+
+A fixture is a picture of one printed system and a reference MusicXML saying
+what that system actually holds, note by note, with MuseScore's own engraving
+coordinates -- that is what lets the fixture matcher line a detected notehead up
+with a printed one without rendering anything.
+
+Both halves come from work already done here: the picture is the band a person
+drew in the Systems editor, cropped straight off the PDF, and the reference is
+the cleaned score imploded back to the shape of the print, trimmed to that band's
+bars.  So the fixture says what the page says, not what any parse said.
+
+    .venv/bin/python scripts/make_stem_fixture.py laulun-aika-3 2 --name laulun-aika-s2
+
+Writes `<HOMR_FIXTURES>/<name>.png` and `.musicxml` (default: the system-4
+worktree's `fixtures/`), and prints the entry to add to
+`stem-direction-fixtures.json`.
+"""
+
+import argparse
+import json
+import os
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+from dotenv import load_dotenv
+from lxml import etree
+
+sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+
+from scripts.implode_report import drop_rests_for, override_for  # noqa: E402
+from scripts.reference_manifest import reference_files  # noqa: E402
+from src.clean_score.implode import implode  # noqa: E402
+from src.song_app import pdf_systems  # noqa: E402
+
+FIXTURES = Path(os.environ.get("HOMR_FIXTURES",
+                               "/var/home/eero/homr-trees/system-4/fixtures"))
+#: What the fixture pictures are cropped at. The existing ones are a few hundred
+#: pixels tall, and homr's staff detection wants the staff lines resolved rather
+#: than the page legible.
+FIXTURE_DPI = 200
+
+#: Carried forward into the trimmed score: a system that does not start the piece
+#: prints no clef, key or meter of its own, but a score that opens there needs
+#: all three or every pitch in the reference is read against the wrong staff.
+CARRIED = ("Clef", "KeySig", "TimeSig")
+
+
+def trim(root: etree._Element, start: int, end: int) -> None:
+    """Keep only bars `start`..`end`, with the state they inherit written in."""
+    for staff in root.find("Score").findall("Staff"):
+        bars = staff.findall("Measure")
+        carried = {}
+        for bar in bars[:start - 1]:
+            for tag in CARRIED:
+                for found in bar.iter(tag):
+                    carried[tag] = found
+        keep = bars[start - 1:end]
+        for bar in bars:
+            if bar not in keep:
+                staff.remove(bar)
+        if not keep:
+            continue
+        voice = keep[0].find("voice")
+        if voice is None:
+            voice = etree.SubElement(keep[0], "voice")
+        for offset, tag in enumerate(CARRIED):
+            if keep[0].find(f".//{tag}") is None and tag in carried:
+                voice.insert(offset, carried[tag])
+
+
+def main() -> None:
+    load_dotenv()
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("slug")
+    parser.add_argument("system", type=int)
+    parser.add_argument("--name", default="")
+    parser.add_argument("--dpi", type=int, default=FIXTURE_DPI)
+    args = parser.parse_args()
+
+    name = args.name or f"{args.slug}-s{args.system}"
+    song_dir = f"songs/{args.slug}"
+    sources = reference_files(args.slug)
+    pdf = str(sources.pdf)
+
+    bands = {b.index: b for b in pdf_systems.load_bounds(song_dir)}
+    band = bands[args.system]
+    if not band.measure_start:
+        raise SystemExit(f"System {args.system} has no measure range; label the "
+                         "bounds against the score first.")
+
+    FIXTURES.mkdir(parents=True, exist_ok=True)
+    with tempfile.TemporaryDirectory() as tmp:
+        image = pdf_systems.crop_systems(pdf, [band], tmp, dpi=args.dpi)[0]
+        picture = FIXTURES / f"{name}.png"
+        picture.write_bytes(Path(image.path).read_bytes())
+
+        root = etree.parse(str(sources.cleaned)).getroot()
+        implode(root, override_for(args.slug), drop_rests_for(args.slug))
+        trim(root, band.measure_start, band.measure_end)
+        score = Path(tmp) / f"{name}.mscx"
+        etree.ElementTree(root).write(str(score), encoding="UTF-8",
+                                      xml_declaration=True)
+        out = FIXTURES / f"{name}.musicxml"
+        cli = os.environ.get("MUSESCORE_CLI_PATH", "musescore3")
+        result = subprocess.run([cli, str(score), "-o", str(out)],
+                                capture_output=True, text=True, timeout=300)
+        if result.returncode != 0 or not out.exists():
+            raise SystemExit(f"MuseScore could not export the reference:\n"
+                             f"{result.stderr[-800:]}")
+
+    print(f"{picture}\n{out}")
+    print("\nAdd to fixtures/stem-direction-fixtures.json:\n")
+    print(json.dumps({name: {"image": picture.name, "reference": out.name,
+                             "known_gaps": []}}, indent=2))
+
+
+if __name__ == "__main__":
+    main()
```

</details>

<details><summary><code>scripts/scan_references.py</code> +102−0</summary>

```diff
diff --git a/scripts/scan_references.py b/scripts/scan_references.py
new file mode 100644
index 0000000..42b30ed
--- /dev/null
+++ b/scripts/scan_references.py
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""Read each reference song's page with homr, into a scratch song of its own.
+
+The six reference songs are finished work -- recorded, uploaded, sung -- and the
+scan stage writes into the song it scans and can put it back on `scan` when a
+band comes back a hole.  So none of this touches them: each gets a throwaway song
+under `scan-eval/` holding a link to its PDF and a copy of its reviewed
+`.systems.json`, and that is what is scanned.  What comes out is the same
+`scanned.musicxml` the app would produce, next to a reference nobody scanned.
+
+**Which homr reads it is the point**, so it is named rather than defaulted, and
+the results are kept under the engine that produced them: the installed venv is
+whatever was frozen there, and a working copy is the fork's branch being tried.
+Two engines over the same songs is the comparison this exists to make.
+
+    .venv/bin/python scripts/scan_references.py --engine system-4
+    .venv/bin/python scripts/scan_references.py --engine default <slug> ...
+
+Writes `scan-eval/<engine>/<slug>/`, and prints a line per system.
+"""
+
+import json
+import os
+import shutil
+import sys
+import time
+from pathlib import Path
+
+from dotenv import load_dotenv
+
+sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+
+from src.song_app import omr, pdf_systems, scan, state  # noqa: E402
+from scripts.reference_manifest import manifest, reference_files  # noqa: E402
+
+SCRATCH = Path("scan-eval")
+
+
+def in_the_set() -> list[str]:
+    listed = manifest()
+    return [name for name, entry in listed.items()
+            if entry["review"]["status"] != "excluded"]
+
+
+def stage_a_copy(slug: str, root: Path) -> None:
+    """A song holding nothing but the page and the bands drawn on it."""
+    real = Path("songs") / slug
+    sources = reference_files(slug)
+    pdf = sources.pdf.name
+    here = root / slug
+    here.mkdir(parents=True, exist_ok=True)
+    target = here / pdf
+    if not target.exists():
+        # A link, not a copy: these are scans of in-copyright editions and one
+        # of them is 4 MB.
+        os.symlink(os.path.abspath(sources.pdf), target)
+    shutil.copy(real / pdf_systems.BOUNDS_FILE, here / pdf_systems.BOUNDS_FILE)
+    # The state file is where the fragments already read are recorded, so
+    # rewriting it is throwing away the scan: the MusicXML stays on disk but
+    # nothing knows it is there, and every band is read again. Bands that really
+    # did move are `reconcile`'s business, not this script's.
+    if (here / ".song.json").exists():
+        return
+    (here / ".song.json").write_text(json.dumps({
+        "name": slug, "stage": "scan", "sources": {"pdf": pdf},
+    }, indent=2, ensure_ascii=False))
+
+
+def main() -> None:
+    load_dotenv()
+    argv = sys.argv[1:]
+    key = omr.DEFAULT_ENGINE
+    if "--engine" in argv:
+        at = argv.index("--engine")
+        key = argv[at + 1]
+        argv = argv[:at] + argv[at + 2:]
+    engine = None if key == omr.DEFAULT_ENGINE else omr.engine_for(key)
+    print(f"reading with: {key} -- "
+          f"{engine.label if engine else omr.engines()[0].label}", flush=True)
+    slugs = argv or in_the_set()
+    root = (SCRATCH / key).resolve()
+    for slug in slugs:
+        stage_a_copy(slug, root)
+
+    # Every song here is one of these throwaways, so nothing real is in reach.
+    state.SONGS_DIR = str(root)
+    for slug in slugs:
+        print(f"\n===== {slug}", flush=True)
+        began = time.time()
+        song = state.load(slug)
+        result = scan.run(song, log=lambda m: print("  " + m, flush=True),
+                          engine=engine)
+        took = time.time() - began
+        holes = result.get("holes") or []
+        print(f"  -> {result['read']} of {result['systems']} systems read, "
+              f"{len(holes)} hole(s) {holes}, "
+              f"{'assembled' if result['complete'] else 'NOT assembled'}, "
+              f"{took:.0f}s", flush=True)
+
+
+if __name__ == "__main__":
+    main()
```

</details>

<details><summary><code>scripts/propose_bounds.py</code> +54−0</summary>

```diff
diff --git a/scripts/propose_bounds.py b/scripts/propose_bounds.py
new file mode 100644
index 0000000..4806aa5
--- /dev/null
+++ b/scripts/propose_bounds.py
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+"""Propose the printed-system bands for a song and save them.
+
+The app deliberately never writes `.systems.json` on its own -- `find-systems`
+hands the proposal to the Systems editor unsaved, so a person drags before
+anything is recorded.  This is the same proposal written straight to the file,
+for songs being prepared as OMR fixtures in bulk; **the drag is still owed**, so
+open each song's Systems tab and correct what the finder got wrong.
+
+    .venv/bin/python scripts/propose_bounds.py <slug> [<slug> ...]
+"""
+
+import os
+import sys
+from pathlib import Path
+
+from dotenv import load_dotenv
+
+sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+
+from src.song_app import pdf_systems, pipeline, state, system_finder  # noqa: E402
+from scripts.reference_manifest import reference_files  # noqa: E402
+
+
+def bounds_score(song) -> str:
+    xml = song.source_path("xml")
+    if not xml or not os.path.exists(xml):
+        return ""
+    try:
+        return pipeline.convert_to_mscx(xml, song.dir)
+    except Exception:
+        return ""
+
+
+def main() -> None:
+    load_dotenv()
+    # The page a fixture is checked against is the manifest's, not whichever
+    # PDF the song folder happens to name first: which one was the printed page
+    # has twice been the thing that was wrong.
+    for slug in sys.argv[1:]:
+        song = state.load(slug)
+        pdf = str(reference_files(slug).pdf)
+        print(f"== {slug}: {os.path.basename(pdf)}", flush=True)
+        bands = system_finder.find_bands(pdf, log=lambda m: print("  " + m, flush=True))
+        saved = pipeline.save_system_bounds(
+            song.dir, [b.to_dict() for b in bands], bounds_score(song)
+        )
+        labelled = sum(1 for b in saved if b.get("measure_start"))
+        print(f"   saved {len(saved)} bands to {pdf_systems.BOUNDS_FILE}"
+              f" ({labelled} labelled with a measure range)", flush=True)
+
+
+if __name__ == "__main__":
+    main()
```

</details>

<details><summary><code>scripts/write_implode_manifest.py</code> +81−0</summary>

```diff
diff --git a/scripts/write_implode_manifest.py b/scripts/write_implode_manifest.py
new file mode 100644
index 0000000..7c71c4a
--- /dev/null
+++ b/scripts/write_implode_manifest.py
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+"""Write down which songs can be OMR fixtures, and what a person said about them.
+
+The songs themselves cannot be committed -- they are gitignored working folders
+holding scans of in-copyright editions -- so what travels is this manifest: for
+each song, the files a fixture is built from, a hash of each so a rebuild can
+prove it used the same music, how its staves map back onto the printed page, and
+the verdict of somebody who looked.
+
+The verdicts are the part that cannot be regenerated.  Everything else this
+script derives; a `review` block is only ever written by hand, and rerunning
+keeps whatever is already there.
+
+    .venv/bin/python scripts/write_implode_manifest.py
+"""
+
+import hashlib
+import json
+import sys
+from pathlib import Path
+
+from lxml import etree
+
+sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
+
+from scripts.implode_report import candidate_songs, override_for  # noqa: E402
+from src.clean_score.implode import grouping  # noqa: E402
+
+MANIFEST = Path("fixtures/omr-songs.json")
+
+
+def digest(path: Path) -> str:
+    return hashlib.sha256(path.read_bytes()).hexdigest()[:16]
+
+
+def main() -> None:
+    existing = json.loads(MANIFEST.read_text()) if MANIFEST.exists() else {}
+    # Both of these are somebody's reading of the page, and nothing here can
+    # derive either: keep whatever is already written down.
+    kept = {
+        name: (
+            entry.get("review"),
+            (entry.get("grouping") or {}).get("override"),
+            entry.get("source_override"),
+        )
+        for name, entry in existing.get("songs", {}).items()
+    }
+
+    out = {}
+    for name, pdf, cleaned, made in candidate_songs(with_excluded=True):
+        found = grouping(etree.parse(str(cleaned)).getroot(), override_for(name))
+        out[name] = {
+            "pdf": pdf.name,
+            "cleaned": cleaned.name,
+            "pdf_sha256": digest(pdf),
+            "cleaned_sha256": digest(cleaned),
+            "video": made,
+            "grouping": {
+                "source": found.source,
+                "inferred": found.inferred,
+                "printed": [printed.staves for printed in found.printed],
+                "labels": [printed.label for printed in found.printed],
+            },
+            # Written by hand after somebody has looked at the reference beside
+            # the page.  Nothing here is derived, and nothing regenerates it.
+            "review": kept.get(name, (None, None, None))[0] or {"status": "unreviewed", "notes": ""},
+        }
+        review, override, source = kept.get(name, (None, None, None))
+        if override:
+            out[name]["grouping"]["override"] = override
+        if source:
+            out[name]["source_override"] = source
+
+    MANIFEST.parent.mkdir(exist_ok=True)
+    MANIFEST.write_text(json.dumps({"version": 1, "songs": out}, indent=2, ensure_ascii=False) + "\n")
+    out_count = sum(1 for entry in out.values() if entry["review"]["status"] == "excluded")
+    print(f"{MANIFEST}: {len(out)} songs, {out_count} excluded, {len(out) - out_count} in the set")
+
+
+if __name__ == "__main__":
+    main()
```

</details>

<details><summary><code>.gitignore</code> +8−0</summary>

```diff
diff --git a/.gitignore b/.gitignore
index b0df4dc..d8d605e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -32,3 +32,11 @@ playlists.txt
 /*.mscx
 /*.mscz
 .venv
+
+# Built by scripts/implode_report.py
+implode-report/
+
+# Built by scripts/song_pages.py
+song-pages/
+scan-eval/
+scan-report/
```

</details>

Not inlined — review these in the Files changed tab:
- `src/clean_score/implode.py` +491−0, 491 diff lines; exceeds the per-file limit.
- `scripts/scan_vs_reference.py` +191−0, 191 diff lines; exceeds the per-file limit.
- `scripts/implode_report.py` +246−0, 246 diff lines; exceeds the per-file limit.
- `scripts/scan_report.py` +200−0, 200 diff lines; exceeds the per-file limit.
- `scripts/song_pages.py` +131−0, 131 diff lines; exceeds the remaining packet limit.

The PR now enforces the reviewed manifest paths and hashes throughout reference work, excludes hidden resting staves from voice totals, and preserves changing `lyricsSystemMap` groups instead of unioning them across the score. Fixed output positions carry each system’s mapped voices, while unused positions hold measure rests and are hidden.

No visual change.

Verification:
- Focused implosion, reference-manifest, and real clean-flow tests: 30 passed.
- The Laulun aika regression verifies T1/T2 share one printed staff in measure 1 and split onto separate printed staves in measure 26.
- Reviewed overrides still beat a changing recorded system map.
- Every current manifest entry matches both recorded hashes.
- Pushed SHA: `a099a9099607346ef0730799b8d103f2a803dd00`.

Review notes: the only cleaned-score discovery left is in `candidate_songs()`, used exclusively by `write_implode_manifest.py` to regenerate the manifest before a new review. The pushed CI run is still in progress.

AgentDeck session link unavailable in this worker environment.



---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
/claude One correctness issue remains in head `a099a9099607346ef0730799b8d103f2a803dd00`: the new fixed-position system implosion can change which source staff occupies a position at a system boundary, but `_merge_system_staves()` does not carry the new source staff's inherited clef/key/time into that boundary.

This matters in the concrete `lyricsSystemMap` shape already covered by `test_clean_flow`: an earlier range can have printed position 2 backed by bass staff 4, while a later range has position 2 backed by T2 staff 2. `_merge_system_staves()` reuses one fixed output staff for position 2. If staff 2's first measure in the later range has no explicit `<Clef>` (normal when it simply inherits treble clef from earlier measures on its original staff), `_merge_measure()` inserts no clef, so the fixed position keeps the bass clef inherited from the previous system. The reference then engraves the later tenor system with the wrong clef/pitches even though the staff split itself is structurally correct.

Please carry `Clef`, `KeySig`, and `TimeSig` state from each source staff into the first measure where that source/group takes over a fixed printed position (similar to the state-carrying logic already used by `make_stem_fixture.trim`). Add a regression where a position changes from an F-clef source to a G-clef source whose boundary measure has no explicit clef, and assert the resulting imploded boundary carries the G clef.

**Comment by eerovil** [comment]
The inherited per-system fix is present at `a099a90`, the worktree is clean, and both prior CI jobs are green. I’ll carry each incoming source staff’s inherited clef, key and meter into the first boundary measure where it takes over a fixed printed position, add the F-clef-to-G-clef regression, run the focused implosion and clean-flow tests, and update PR #138.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
PR #138 was merged while the boundary-state edge-case review was still running. The first fix is now on `main`, but the review found that a resting leading member could donate the wrong clef; I have corrected that ownership rule and will open a focused follow-up PR from the updated base.

---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #138](https://github.com/eerovil/musescore-choir-plugins/pull/138) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)

**Comment by eerovil** [comment]
Review surface: source +48−14; tests +40−3; docs +0−0; generated +0−0. Read first: `src/clean_score/implode.py`.

Beyond the card: nothing

PR #139 carries the final correction after PR #138 merged during edge-case review. Fixed printed positions now take inherited clef, key, and meter from the first source voice that is actually printed. Resting mapped members cannot donate stale state, silent groups do not gain a synthetic voice, and unchanged state is not repeated.

No visual change.

Verification:
- `AGENTDECK_HEAVY_SLOT_HELD=1 .venv/bin/python -m pytest src/clean_score/tests/test_implode.py src/clean_score/tests/test_reference_manifest.py src/song_app/tests/test_clean_flow.py -q` — 33 passed at `0ff3e24f3dff930817e2d58bd041b1ebccddd063`.
- GitHub Actions run [33841270412](https://github.com/eerovil/musescore-choir-plugins/actions/runs/33841270412) was pending when handed off.

Review note: the real Laulun aika clean-flow regression asserts the output’s second fixed position changes from bass F clef to inherited tenor G8vb at the actual system boundary. Synthetic coverage also pins key/time transfer, a resting leading member, and unchanged-state suppression.

<details>
<summary><code>src/clean_score/implode.py</code> +48−14</summary>

```diff
diff --git a/src/clean_score/implode.py b/src/clean_score/implode.py
index 7308372..92a7690 100644
--- a/src/clean_score/implode.py
+++ b/src/clean_score/implode.py
@@ -419,6 +419,17 @@ def _merge_measure(
     return measure
 
 
+def _first_kept_source(
+    sources: list[etree._Element], drop_when_resting: set[int]
+) -> int | None:
+    """Return the source member whose voice carries this printed staff's state."""
+    for position, source in enumerate(sources):
+        for voice in source.findall("voice"):
+            if position not in drop_when_resting or voice.find("Chord") is not None:
+                return position
+    return None
+
+
 def _rest_measure(source: etree._Element) -> etree._Element:
     """A full-bar rest carrying the source measure's staff-level state."""
     measure = etree.fromstring(etree.tostring(source))
@@ -449,19 +460,37 @@ def _source_states(
     return states
 
 
+def _measure_state(measure: etree._Element) -> dict[str, etree._Element]:
+    """Return explicit staff-level state written in one output measure."""
+    return {
+        element.tag: element
+        for element in measure.iter()
+        if element.tag in STAFF_LEVEL
+    }
+
+
 def _carry_source_state(
-    measure: etree._Element, state: dict[str, etree._Element]
+    measure: etree._Element,
+    source_state: dict[str, etree._Element],
+    output_state: dict[str, etree._Element],
 ) -> None:
-    """Write inherited source state when a fixed output position changes source."""
+    """Write changed inherited state when a fixed output position changes source."""
     voice = measure.find("voice")
     if voice is None:
-        voice = etree.SubElement(measure, "voice")
-    present = {element.tag for element in measure.iter() if element.tag in STAFF_LEVEL}
-    insert_at = 0
+        return
+    present = _measure_state(measure)
+    inherited = []
     for tag in STAFF_LEVEL:
-        if tag not in present and tag in state:
-            voice.insert(insert_at, etree.fromstring(etree.tostring(state[tag])))
-            insert_at += 1
+        incoming = source_state.get(tag)
+        previous = output_state.get(tag)
+        if (
+            tag not in present
+            and incoming is not None
+            and (previous is None or etree.tostring(incoming) != etree.tostring(previous))
+        ):
+            inherited.append(incoming)
+    for element in reversed(inherited):
+        voice.insert(0, etree.fromstring(etree.tostring(element)))
 
 
 def _merge_system_staves(
@@ -496,7 +525,8 @@ def _merge_system_staves(
         for element in template:
             if element.tag != "Measure" and not (position > 1 and element.tag == "VBox"):
                 staff.append(etree.fromstring(etree.tostring(element)))
-        previous_group: tuple[int, ...] | None = None
+        previous_owner: int | None = None
+        output_state: dict[str, etree._Element] = {}
         for index, base_measure in enumerate(base_bars):
             group = by_measure[index].printed.get(position)
             if group:
@@ -507,18 +537,22 @@ def _merge_system_staves(
                     if names.get(source, "") in set(drop_rests)
                 }
                 measure = _merge_measure(sources, silent)
-                current_group = tuple(group.staves)
-                if current_group != previous_group:
+                owner = _first_kept_source(sources, silent)
+                current_owner = group.staves[owner] if owner is not None else None
+                if current_owner is not None and current_owner != previous_owner:
                     _carry_source_state(
-                        measure, source_states[group.staves[0]][index]
+                        measure,
+                        source_states[current_owner][index],
+                        output_state,
                     )
             else:
                 measure = _rest_measure(base_measure)
-                current_group = None
+                current_owner = None
             for layout_break in measure.findall("LayoutBreak"):
                 measure.remove(layout_break)
             staff.append(measure)
-            previous_group = current_group
+            output_state.update(_measure_state(measure))
+            previous_owner = current_owner
         merged.append(staff)
 
     top = merged[0].findall("Measure") if merged else []
```
</details>



---
🤖 **Steer this card with a new comment**
- **Reply, continue, or ask for changes:** `/claude <what you want>`
- **Stop the work:** `/claude stop`
- **Approve and release finished work:** `/merge` *(only from In review)*
- **Review before releasing:** `/merge review`

[PR #139](https://github.com/eerovil/musescore-choir-plugins/pull/139) · [Issue #138](https://github.com/eerovil/musescore-choir-plugins/issues/138)
